File: services/payment-service/internal/infrastructure/messaging/rabbitmq/publisher.go
package rabbitmq

import (
	"encoding/json"
	"log"

	amqp "github.com/rabbitmq/amqp091-go"
)

type Publisher struct {
	channel *amqp.Channel
}

func NewPublisher(ch *amqp.Channel) *Publisher {
	return &Publisher{channel: ch}
}

func (p *Publisher) Publish(queue string, body any) error {
	payload, err := json.Marshal(body)
	if err != nil {
		return err
	}

	err = p.channel.Publish(
		"", queue, false, false, 
		amqp.Publishing{
			ContentType: "application/json",
			DeliveryMode: amqp.Persistent,
			Body: payload,
		},
	)

	if err != nil {
		log.Fatalf("Erro ao publicar na fila \"%v\": %v", queue, err)
	}

	log.Printf("Evento publicado na fila \"%v\"", queue)
	return nil
}

File: services/pedido/app/infrastructure/messaging/rabbitmq/publisher.py
import json
import logging
import aio_pika
from aio_pika import Channel

logger = logging.getLogger(__name__)


class Publisher:
    def __init__(self, channel: Channel):
        self._channel = channel

    async def publish_order_created(self, body: dict):
        exchange = await self._channel.get_exchange("order.created.exchange")
        payload = json.dumps(body).encode()

        await exchange.publish(
            aio_pika.Message(
                body=payload,
                content_type="application/json",
                delivery_mode=aio_pika.DeliveryMode.PERSISTENT,
            ),
            routing_key="",  # fanout ignora routing_key
        )

        logger.info('Evento publicado no exchange "order.created.exchange"')